enhancements/network/netflow.md

```diff
@@ -0,0 +1,121 @@
+
+### Non-Goals
+
+- This document only applies to OVN-Kubernetes, not OpenShift SDN or any other other network plugin.
+- This document will not discuss the flow logs store solutions. Typically, customers store the NetFlow data in a data warehouse system (e.g. ClickHouse) or search and analytics system (e.g. ElasticSearch). It is out of the scope of this document to discuss which option should be used and if we should manage that solution.
+- This document will not discuss the aggregation of flow data with corresponding K8S pods.
+- This document will not discuss flow visualization or flow dashboards.
+
+## Proposal
+
+OVS supports NetFlow v5, IPFIX and sflow as export flows protocol. It also supports a list of collectors by specifying IP and port to send the flow data.
+As such, we would support the three protocols and an arbitrary list of collectors.
+
+The Cluster Network Operator (CNO) would expose:
+* *exportNetworkFlowsConfig* dict to contain the options to export the network flows
+  * *exportNetworkFlowsProtocol* to specify which protocol should be used. Valid values should be *netflow*, *ipfix* and *sflow*.
+  * *exportNetworkFlowsCollectors* list to contain the collectors IPs and ports
+    * *exportNetworkFlowsCollectorIP* to specify the IP of the collector that will consume the flow data
+    * *exportNetworkFlowsCollectorPort* to specify the port of the collector that will consume the flow data
+
+This is an example of a *networks.operator.openshift.io* spec containing two collectors:
+
+```yaml
+spec:
+  clusterNetwork:
+  - cidr: 10.128.0.0/14
+    hostPrefix: 23
+  defaultNetwork:
+    ovnKubernetesConfig:
+      exportNetworkFlowsConfig:
+        exportNetworkFlowsCollectors:
+        - exportNetworkFlowsCollectorIP: 172.30.158.150
+          exportNetworkFlowsCollectorPort: 2056
+        - exportNetworkFlowsCollectorIP: 172.30.54.103
+          exportNetworkFlowsCollectorPort: 2056
+        exportNetworkFlowsProtocol: netflow
+      genevePort: 6081
+      mtu: 1400
+    type: OVNKubernetes
+  deployKubeProxy: false
+  disableMultiNetwork: false
+  disableNetworkDiagnostics: false
+  logLevel: Normal
+  managementState: Managed
+  observedConfig: null
+  operatorLogLevel: Normal
+  serviceNetwork:
+  - 172.30.0.0/16
+```
+
+Under the covers, these options will make the CNO to perform an ovs-vsctl command on the bridge br-int, which is the bridge the containers are connected to on OVN-Kubernetes, and its management port ovn-k8s-mp0.
+As an example, if netflow was enabled per the above example this is the command that would be executed.
+
+`ovs-vsctl -- --id=@netflow create netflow targets=\[\"172.30.158.150:2056\",\"172.30.54.103:2056\"\] -- set bridge br-int netflow=@netflow`
+
+This would make OVS to send the flow data to both collectors 172.30.158.150 and 172.30.54.103 on port 2056 by using netflow protocol.
+
+### Test Plan
+
+- Unit tests for the feature
+- e2e tests covering the feature
```

Comment:
Test plan is pretty vague; do we even know what software we could use to act as the collector?

Reply:
Yeah, I guess we can just leverage goflow, since it has a docker image in the hub and is easy to use. Will add.

Reply:
done